packages/components/src/site-thumbnail/index.tsx

```diff
@@ -0,0 +1,47 @@
+import classnames from 'classnames';
+import { ReactNode } from 'react';
+import './style.scss';
+import { useMshotsImg } from './use-mshots';
+import { getTextColorFromBackground } from './utils';
+
+const MSHOTS_OPTION = {
+	vpw: 1200,
+	vph: 1200,
+	w: 300,
+};
+
+type Props = {
+	backgroundColor?: string;
+	mShotsUrl?: string;
+	size?: 'small' | 'medium';
+	children?: ReactNode;
+};
+
+export const SiteThumbnail = ( {
+	backgroundColor,
+	children,
+	mShotsUrl,
+	size = 'small',
+}: Props ) => {
+	const maybeImage = useMshotsImg( mShotsUrl ?? '', MSHOTS_OPTION );
+
+	const src: string = maybeImage?.src || '';
+	const visible = !! src && mShotsUrl;
+	const textColor = backgroundColor && getTextColorFromBackground( backgroundColor );
+
+	const className = classnames(
+		'site-thumbnail',
+		visible ? 'site-thumbnail-visible' : '',
+		`site-thumbnail__size-${ size }`
+	);
+
+	const loader = mShotsUrl ? 'site-thumbnail-loader' : '';
+
+	return ! visible ? (
+		<div className={ className } style={ { backgroundColor: backgroundColor, color: textColor } }>
+			<div className={ loader }>{ children }</div>
+		</div>
+	) : (
+		<img className={ className } src={ src } alt="site thumbnail" />
```

Comment:
Would it make sense to use the same container element for both states, and then swap out the child element when the image is available?

Reply:
Also, for the `<img>`, we may want to generate multiple mShots widths so we can add a `srcset` tag computed from the `size`.

I think the `srcset` tag could be a later enhancement though. For now, we'll need to make sure we generate a sufficiently large image.

Reply:
Thanks for the comments @danielbachhuber! 
> Would it make sense to use the same container element for both states, and then swap out the child element when the image is available?

We should be able to do this,

> Also, for the <img>, we may want to generate multiple mShots widths so we can add a srcset tag computed from the size.
> 
> I think the srcset tag could be a later enhancement though. For now, we'll need to make sure we generate a sufficiently large image.

I think this will need further exploration especially generating multiple widths. This [P2](pbxlJb-1sw-p2) provides further context as to mshots performance in generating 3 different previews. 
Just a note, stumbled on this package https://github.com/thekevinscott/UpscalerJS which uses machine learning to maintain the fidelity of images when scaled.